pytorch_lightning/plugins/precision/apex_amp.py

```diff
@@ -34,24 +35,23 @@ def __init__(self, amp_level: str = "O2") -> None:
         super().__init__()
         self.backend = AMPType.APEX
         self.amp_level = amp_level
+        self._connected = False
 
     def master_params(self, optimizer: Optimizer) -> _PARAMETERS:
         return amp.master_params(optimizer)
 
-    def connect(
-        self,
-        model: Module,
-        optimizers: List[Optimizer],
-        lr_schedulers: List[Any],
-    ) -> Tuple[Module, List[Optimizer], List[Any]]:
-        """Connects the precision plugin to the training process,
-        configures apex and reinits the schedulers
-        """
-        if model.device.type != "cuda":
-            return model, optimizers, lr_schedulers
-        model, optimizers = self.configure_apex(amp, model, list(optimizers), self.amp_level)
-        self.reinit_scheduler_properties(optimizers, lr_schedulers)
-        return model, optimizers, lr_schedulers
+    def pre_dispatch(self) -> None:
+        return super().pre_dispatch()
+
+    def dispatch(self, trainer: "pl.Trainer") -> None:
+        if not self._connected:
+            accelerator = trainer.accelerator
+            model, optimizers = self.configure_apex(accelerator.lightning_module, accelerator.optimizers)
```

Comment:
```suggestion
            model, optimizers = self.configure_apex(trainer.lightning_module, accelerator.optimizers)
```
Let's access it through the trainer like we do for the other plugins?

Reply:
@tchaton n00b question: would we be able to use for apex?

https://github.com/PyTorchLightning/pytorch-lightning/blob/338f5a331155d61a9d94168480ca45c264821019/pytorch_lightning/plugins/training_type/training_type_plugin.py#L193-L201

Reply:
Done !